### PR TITLE
add type hints

### LINKED ISSUE #108: `Report` additional and core argument type
[In the doctring](https://github.com/banesullivan/scooby/blob/main/scooby/report.py#L192-L196) additional and core are written with the type `list[str]|list[ModuleType]`. 
Is it possible to pass an argument of type `list[str|ModuleType]` ?

Is it also possible to add type hints for the library ?

### PR DESCRIPTION
**Pull request by Gryfenfer97**
add type hints

I am still not sure about the type of `extra_meta` in `Report` because there is check about it being a list.

Should fix #108

### CODE REVIEW

File: scooby/knowledge.py [hunk 1]
```diff
@@ -190,7 +187,7 @@ def meets_version(version, meets):
     return True
 
 
-def get_filesystem_type():
+def get_filesystem_type() -> str | Literal[False]:
```
Comment (Gryfenfer97) [inline scooby/knowledge.py:190]: Should use Union to be python<3.10 compliant

**Comment by Gryfenfer97** [comment]
@banesullivan I still need to import Literal from `typing_extension` for python 3.7.
However this version is no longer supported so if you still want to support it, tell me so I can add the correct import

I also recommend you to use a static type checker when looking at the code as it reveal some problems:
ex [L363 of report.py](https://github.com/banesullivan/scooby/blob/74bb909415b46b5458937adef4bbb91b6672c5fc/scooby/report.py#L363) where `self._extra_meta` is defined as Optional and so, can be None
```py
for meta in self._extra_meta:
    html, i = cols(html, meta[1], meta[0], self.ncol, i)
```

**Comment by banesullivan** [comment]
@Gryfenfer97, apologies for my delay in addressing this. I'm curious if you'd still like to land these changes? There appears to be a few CI testing issues

**Comment by Gryfenfer97** [comment]
> @Gryfenfer97, apologies for my delay in addressing this. I'm curious if you'd still like to land these changes? There appears to be a few CI testing issues

The issue in Python 3.8 was not intentional. Fixed now.
My question was for python 3.7, do you still want to support it or not ? If yes I have to add `typing_extension` as a dependency.

**Comment by banesullivan** [comment]
Ah, let me see if we can drop support for 3.7

**Comment by banesullivan** [comment]
Given 3.7 is past end of life and PyVista (the biggest downstream dependency) has also dropped 3.7, we are good to remove it. Would you like me to handle that in a separate PR or would you be up for handling it here?

**Comment by Gryfenfer97** [comment]
I think it would make more sense in the git history to have a separate commit for it. So doing it in a separate PR would be better for me

**Comment by Gryfenfer97** [comment]
I hope this commit fix all the issues

By the way, there is a bot that can apply pre-commit automatically.

**Comment by banesullivan** [comment]
I'm aware of the pre-commit bot, but this repo is generally pretty low maintenance so I haven't wanted the overhead of adding it.

Looks like 3.8 is still failing on CI for some reason

**Comment by Gryfenfer97** [comment]
> Looks like 3.8 is still failing on CI for some reason

I hope this one fix everything

**Review by banesullivan — APPROVED** [review]
Thanks for these additions and your patience!

**Pull request merged**